framework/SwiftPlot/Axis.swift

```diff
@@ -1,10 +1,10 @@
-public class Axis{
+public class Axis<T,U>{
     public var scaleX: Float = 1
     public var scaleY: Float = 1
```

Comment:
For scaleX and scaleY, should they be generic in the types of the Series? Type T for x, type U for y.

If that's the case, you won't be able to assign a default value. That would then imply that these should be optional, with nil as a default value, so that they can be ignored when no scaling is desired.

Reply:
I don't think scaleX and scaleY need to be generics. scales used everywhere are always Floats.

Reply:
OK, worst case, if you do need this to be generic at some point you should be able to go back and change this then.